tiledb/sm/misc/utils.cc

```diff
@@ -449,6 +449,39 @@ uint64_t common_prefix_size(const std::string& a, const std::string& b) {
   return size;
 }
 
+std::string rest_components_from_url(const std::string& url) {
+  std::string path_delimiter = "/";
+  std::vector<std::string> path_parts;
+  size_t start;
+  size_t end = 0;
+  while ((start = url.find_first_not_of(path_delimiter, end)) !=
+         std::string::npos) {
+    end = url.find(path_delimiter, start);
+    path_parts.emplace_back(url.substr(start, end - start));
+  }
+
+  std::string array_ns;
+  std::string array_uri;
+  // Find rest components array_ns and array_uri
+  for (std::vector<std::string>::iterator part_it = path_parts.begin();
+       part_it != path_parts.end();
+       ++part_it) {
+    if (*part_it == "arrays") {
+      // Get the next two records that refer to array_ns and array_uri
+      // accordingly
+      if (++part_it != path_parts.end() && ++(++part_it) != path_parts.end()) {
+        array_ns = *(++part_it);
+        array_uri = *(++part_it);
```

Comment:
The prefix iterators on L472 mutate `part_it`, so L473 and L474 will return the 3rd and 4rth elements past "arrays".

Refactor L463-L482
```
for (size_t i = 0; i < path_parts.size(); ++i) {
  if (path_parts[i] == "arrays") {
    if (i + 2 < path_parts.size()) {
      return path_parts[i + 1] + ":" + path_parts[i + 2];
    }
  }
}

return "";
```